### PR TITLE
Enable Flexible Spark Versions in Release Shell Script

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

Enable Flexible Spark Versions in Release Shell Script
Simplify our different releasing scripts in case of spark2.x or spark3.x 

### 1. Why the change?

Provide users with a customized way of packaging bigdl with diffierent spark version, and refine some details.

### 2. User API changes

image

Usage: release_default_linux_spark.sh version quick_build upload **spark_version** suffix mvn_parameters
Usage example: bash release_default_linux_spark.sh default false true **3.1.2** true
Usage example: bash release_default_linux_spark.sh 0.14.0.dev1 false false **2.4.6** false -Ddata-store-url=..

### 3. Summary of the change 

1. Replace `release_default_linux_spark246.sh` and `release_default_linux_spark312.sh` under each submodule with only a `release_default_linux_spark.sh `
2. Support **spark_version** parameter to `release_default_linux_spark.sh `
3. Refine some details.

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

**Comment by leonardozcm** [comment]
http://10.112.231.51:18888/job/BigDL-Orca-PR-Validation/870/
http://10.112.231.51:18888/job/ZOO-PR-BigDL-Python-Spark-3.1-py36-ray/911/

**Comment by leonardozcm** [comment]
I will apply these changes on mac related scripts in another pr.

File: docs/readthedocs/source/doc/UserGuide/develop.md [hunk 1]
```diff
@@ -19,8 +19,7 @@ To generate a new [whl](https://pythonwheels.com/) package for pip install, you
 
 ```bash
 cd BigDL/python/dev
-bash release_default_linux_spark246.sh default false false false  # build on Spark 2.4.6 for linux
-# Use release_default_linux_spark312.sh to build on Spark 3.1.2 for linux
+bash release_default_linux_spark.sh default false false 2.4.6 false  # build on Spark 2.4.6 for linux, and specify spark_version at 3.1.2 to build on Spark 3.1.2 for linux
 # Use release_default_mac_spark246.sh to build on Spark 2.4.6 for mac
 # Use release_default_mac_spark312.sh to build on Spark 3.1.2 for mac
```
Comment (leonardozcm) [inline docs/readthedocs/source/doc/UserGuide/develop.md:24]: mac

**Comment by leonardozcm** [comment]
> I will apply these changes on mac related scripts in another pr.

Schedule canceled, for we can build whls for linux or mac platform with only one script.

**Comment by leonardozcm** [comment]
http://10.112.231.51:18888/job/BigDL-Orca-PR-Validation/872/

File: python/dev/release_default_spark.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+FRIESIAN_SCRIPT_DIR="$(cd ${BIGDL_DIR}/python/friesian/dev/release; pwd)"
+echo $FRIESIAN_SCRIPT_DIR
+bash ${FRIESIAN_SCRIPT_DIR}/release_default_spark.sh ${platform} ${version} true ${upload} ${spark_version} ${suffix}
+
+CHRONOS_SCRIPT_DIR="$(cd ${BIGDL_DIR}/python/chronos/dev/release; pwd)"
+echo $CHRONOS_SCRIPT_DIR
+bash ${CHRONOS_SCRIPT_DIR}/release_default_spark.sh ${platform} ${version} ${upload} ${spark_version} ${suffix}
+
+bash ${RUN_SCRIPT_DIR}/release_bigdl_spark${spark_first_version}.sh linux ${version} ${upload} ${suffix}
```
Comment (hkvision) [inline python/dev/release_default_spark.sh:79]: linux should be platform?

**Review by hkvision — APPROVED** [review]

**Comment by leonardozcm** [comment]
http://10.112.231.51:18889/job/BigDL2.0-PR-Deploy-PYTHON-PyPI-all/7/

**Pull request merged**